src/plugins/intel_npu/src/plugin/src/compiler_adapter.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "compiler_adapter.hpp"
+
+#include "compiler.hpp"
+#include "driver_compiler_adapter.hpp"
+
+namespace intel_npu {
+
+std::vector<uint8_t>& CiDGraph::getCompiledNetwork() {
+    auto cidCompiler = std::dynamic_pointer_cast<driverCompilerAdapter::LevelZeroCompilerAdapter>(_compiler._ptr);
+    cidCompiler->getCompiledNetwork(_graphHandle, _compiledNetwork);
+    return _compiledNetwork;
+}
+
+CompilerAdapter::CompilerAdapter(std::shared_ptr<NPUBackends> npuBackends, ov::intel_npu::CompilerType compilerType)
+    : _compilerType(compilerType),
+      _logger("CompilerAdapter", Logger::global().level()) {
+    _logger.debug("initialize CompilerAdapter start");
+    _compiler = createCompiler(npuBackends, compilerType);
+}
+
+uint32_t CompilerAdapter::getSupportedOpsetVersion() const {
+    return _compiler->getSupportedOpsetVersion();
+}
+
+std::shared_ptr<IGraph> CompilerAdapter::compile(const std::shared_ptr<const ov::Model>& model,
```

Comment:
CompilerAdapter duplicates with ICompiler. If we keep special API for each compiler, the compile() and query() is useless  for DriverCompiler. It is not a good idea to bring CompilerAdapter since we let compiler details to limit definition of the interface